tests/test_filteredtree.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Liblarch - a library to handle directed acyclic graphs
+# Copyright (c) 2011-2012 - Lionel Dricot & Izidor Matušov
+#
+# This program is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+
+import unittest
+from liblarch.treenode import _Node
+from liblarch.tree import MainTree
+from liblarch.filters_bank import FiltersBank
+from liblarch.filteredtree import FilteredTree
+
+
+class TestFilteredTree(unittest.TestCase):
+    def setUp(self):
+        self.added_node = 0
+        self.deleted_node = 0
+        self.tree = MainTree()
+        self.filtersbank = FiltersBank(self.tree)
+        self.filtered_tree = FilteredTree(self.tree, self.filtersbank)
+        self.tree.add_node(_Node(node_id="apple"), priority="high")
+        self.tree.add_node(_Node(node_id="google"), priority="high")
+
+        self.filtered_tree.set_callback('deleted', self.deleted)
+        self.filtered_tree.set_callback('added', self.added)
+
+    def search_filter(self, node, parameters):
+        return node.get_id() == parameters['node_id']
+
+    def true_filter(self, node):
+        return True
+
+    def test_filteredTree_refresh_with_parameters(self):
+        self.filtersbank.add_filter("search_filter", self.search_filter)
+
+        self.filtered_tree.apply_filter("search_filter",
+                                        parameters={'node_id': 'apple'})
+        self.assertTrue(self.filtered_tree.is_displayed(node_id="apple"))
+        self.assertFalse(self.filtered_tree.is_displayed(node_id="google"))
+
+        self.assertEqual(1, self.added_node)
+        self.assertEqual(2, self.deleted_node)
```

Comment:
Can you add a comment here that when refiltering happens, all nodes are removed and only nodes that stay are added back? To explain why you have 2 delted nodes